pkg/messages/baseMessage_test.go

```diff
@@ -0,0 +1,96 @@
+package messages
+
```

Comment:
Should we create a `test` package?

Reply:
I think we should be testing those components in a more connected way? like send a message and then receive it and then check it's been received correctly / to the right agent?

cc @jzzheng22

Reply:
also @woonmoon I think in go that's how tests are done (you add a _test file in the same package), but I might be wrong

Reply:
Ideally we would test the message passing properly... but I'm just getting the easy one here for now. Can create an issue for longer testing... but also only two weeks left.